tests/filesystem/UploadTest.php

```diff
@@ -84,15 +84,92 @@ public function testAllowedFilesize() {
 			'error' => UPLOAD_ERR_OK,
 		);
 		
+		// test upload into default folder
+		$u1 = new Upload();
 		$v = new UploadTest_Validator();
+		
 		$v->setAllowedMaxFileSize(array('txt' => 10));
+		$u1->setValidator($v);
+		$result = $u1->load($tmpFile);
+		$this->assertFalse($result, 'Load failed because size was too big');
 		
-		// test upload into default folder
-		$u1 = new Upload();
+		$v->setAllowedMaxFileSize(array('[doc]' => 10));
+		$u1->setValidator($v);
+		$result = $u1->load($tmpFile);
+		$this->assertFalse($result, 'Load failed because size was too big');
+		
+		$v->setAllowedMaxFileSize(array('txt' => 200000));
 		$u1->setValidator($v);
 		$result = $u1->load($tmpFile);
+		$this->assertTrue($result, 'Load failed with setting max file size');
+		
+		// check max file size set by app category
+		$tmpFileName = 'UploadTest-testUpload.jpg';
+		$tmpFilePath = TEMP_FOLDER . '/' . $tmpFileName;
+		file_put_contents($tmpFilePath, $tmpFileContent . $tmpFileContent);
+		
+		$tmpFile = array(
+			'name' => $tmpFileName,
+			'type' => 'image/jpeg',
+			'size' => filesize($tmpFilePath),
+			'tmp_name' => $tmpFilePath,
+			'extension' => 'jpg',
+			'error' => UPLOAD_ERR_OK,
+		);
 		
+		$v->setAllowedMaxFileSize(array('[image]' => '40k'));
+		$u1->setValidator($v);
+		$result = $u1->load($tmpFile);
+		$this->assertTrue($result, 'Load failed with setting max file size');
+		
+		$v->setAllowedMaxFileSize(array('[image]' => '1k'));
+		$u1->setValidator($v);
+		$result = $u1->load($tmpFile);
 		$this->assertFalse($result, 'Load failed because size was too big');
+		
+		$v->setAllowedMaxFileSize(array('[image]' => 1000));
+		$u1->setValidator($v);
+		$result = $u1->load($tmpFile);
+		$this->assertFalse($result, 'Load failed because size was too big');
+		
+		// Check the max file size overrides the config
+		$maxFileSizes = array(
+			'[image]' => '1k',
+			'txt' => 1000
+		);
+		Config::inst()->update('Upload_Validator', 'default_max_file_size', $maxFileSizes);
+		$v = new UploadTest_Validator();
+		
+		$retrievedSize = $v->getAllowedMaxFileSize('[image]');
+		$result = $retrievedSize == File::ini2bytes($maxFileSizes['[image]']);
```

Comment:
By using `File::ini2bytes` you're relying on the exact same logic as the function you're testing. this should be a hardcoded 1000